app/src/main/java/com/sanjeevyadavit/magecart/MainActivity.kt

```diff
@@ -1,43 +1,94 @@
 package com.sanjeevyadavit.magecart
 
+import android.content.Context
+import android.content.SharedPreferences
 import android.os.Bundle
-import androidx.activity.ComponentActivity
-import androidx.activity.compose.setContent
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.material.MaterialTheme
-import androidx.compose.material.Surface
-import androidx.compose.material.Text
-import androidx.compose.runtime.Composable
-import androidx.compose.ui.Modifier
-import androidx.compose.ui.tooling.preview.Preview
-import com.sanjeevyadavit.magecart.ui.theme.MageCartTheme
-
-class MainActivity : ComponentActivity() {
+import android.view.View
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.navigation.NavController
+import androidx.navigation.fragment.NavHostFragment
+import androidx.navigation.ui.AppBarConfiguration
+import androidx.navigation.ui.setupWithNavController
+import com.sanjeevyadavit.magecart.databinding.ActivityMainBinding
+import com.sanjeevyadavit.magecart.ui.LoginFragment
+import com.sanjeevyadavit.magecart.viewmodel.MainActivityViewModel
+
+class MainActivity : AppCompatActivity() {
+
+    private val viewModel: MainActivityViewModel by viewModels()
+    private lateinit var navController: NavController
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var sharedPref: SharedPreferences
```

Comment:
instead of doing this, you can also create a singleton for sharedPref and get it anywhere you want. SharedPref also maintains singleton at their end, but having it initialised in the project at single place and using that instance everywhere is a good practice. DI will ease this whole process.

Reply:
Yes, Will create singleton for SharedPreferences